### ISSUE TITLE
zmax set incorrectly in rtheta import model

### ISSUE THREAD

**Issue by Higginbottom** [sirocco-rt/sirocco#571]
I've been trying to set up a simple shell model to test the flux calculations in python. All was going reasonably well (apart from the tricky nature of making an imported model) but when the thickness of the shell exceeded a certain value I get this flux vs angle (should be flat)

![Flux_by_bin](https://user-images.githubusercontent.com/3329213/62366546-1c01bb80-b51f-11e9-93e1-921c023447c5.png)

After some investigation, I discover that although the shell is 1e11cm thick, and has an inner edge at 1e11cm, zmax is being set to 1e11. rho_max is 1e12 and rmax is 1e12 as should be. Compled with the fact that smax ends up set to 0.5e11 what happens is that photons are forced to scatter midway through the shell - they go straight on which is fine (density is very low) but for those in more play bins, midway through the shell pushes their z component above zmax and so they are lost. Hence the factor of 2 drop for polar cells - the photons only do 50% of the path through the cell before they are deemed to be outside wind.

I see where this is done (in rtheta_make_grid_import in import_rtheta.c) - just working out the correct logic and then will fix.

**Comment by Higginbottom** [comment]
So, the answer seems too simple - I'm concerned I've misunderstood the logic.

When we read in the grid - the i number is for the radial cell number, and the j is for theta cell number. Once the grid is read in, ndim is set to the number of radial cells, and mdim is set to the number of theta cells.

So, when we come to rtheta_make_grid_import we first loop over all the cells we have read in. The inner coordinate of each cell is set to w[nn].x[0],w[nn].x[1],w[nn].x[2].

There are now a couple of lines used in domains - which I think uses the wind_x variable to store the radial cell boundaries, and the wind_z variable to store the theta cell boundaries.

Then the maximum z,rho and r extent of the wind is computed. This is used to work out if photons have left the wind during transport. 

So we loop over all the cells once more. 
I think this loop is intended to loop over all the theta cells at a given radius, then the next theta shell out etc. 

so r_inner is initially just set to length (w[nn].x)

then nn_outer is set to nn + xx_rtheta.mdim - which is the cell one cell radially outwards.

now, if we aren't in the last radial shell checked by ( if (nn_outer < zdom[ndom].ndim2). )

we set r_outer to the length of w[nn_outer].x.

Ok, we now test if the cell is in the wind, if so the first check is

if (w[nn_outer].x[0] > rho_max) in other words is the inner x coordinate of the next shell out larger than the current rho max - if so it is reset.

Now the problem, the next test is

 if (w[nn - 1].x[2] > zmax)   this is the test that makes the mistake - this is asking if the inner z coordinate of the *previous* cell is larger than the current value of zmax. This means that at best, zmax ends up set to the inner z coordinate of the current shell. It seems to me that this should simply be

if (w[nn_outer].x[2] > zmax) - i.e. just like the test for rhomax. if this test is passed, then just like rho max we set zmax to the inner z coordinate of the next shell out. 

But this seems too simple - and makes me worry that the original code had something else in mind. Would appreciate a quick look and check that nothing different was intended....

**Comment by kslong** [comment]
I can imagine this is a problem that may not have been obvious because likely the tests all involved items with more dimensions, but @Higginbottom  can you provide the example that generated the problem here, so I can see if for myself.

**Comment by Higginbottom** [comment]
Here is the .pf file:

[shell2.pf.txt](https://github.com/agnwinds/python/files/3472878/shell2.pf.txt)

and the model file


[model.dat.txt](https://github.com/agnwinds/python/files/3472880/model.dat.txt)

**Comment by kslong** [comment]
@Higginbottom   I have finally looked at this, after having spend some time on other things I thought might be related.  My take is that several of these are messed up, although it only becomes glaringly apparent when you have a small number of valid cells as in the import grid.  The drawing below indicates where I think one should take the various parameters from if a given cell nn is in the grid in an imported rtheta model.

![IMG_1607](https://user-images.githubusercontent.com/2530901/62736523-232f3a80-b9f3-11e9-86cd-b8594e8c81a3.JPG)

**Comment by kslong** [comment]
I have made modifications along these lines in branch bug571_zmax.  Having done that this is what I get for the boundaries:

```
Imported:    rmin    rmax  1.000000e+11 2.000000e+11
Imported:    zmin    zmax  6.123234e-06 2.000000e+11
Imported: rho_min rho_max  0.000000e+00 2.000000e+11
```

**Comment by kslong** [comment]
@Higginbottom  I have implemented this changes in #582 and requested that you review it. You might want to check that your model not produces exactly what you expected.  When you are happy, you should squash and merge the branch into dev, and then close this issue.
